src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,87 +1,61 @@
 package core.basesyntax;
 
 import java.util.NoSuchElementException;
-import java.util.Objects;
 
 public class ArrayList<T> implements List<T> {
-    private static final int STANDARD_CAPACITY = 10;
+    private static final int DEFAULT_CAPACITY = 10;
     private T[] arrayList;
     private int size;
 
     public ArrayList() {
-        arrayList = (T[]) new Object[STANDARD_CAPACITY];
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
     }
 
     @Override
     public void add(T value) {
-        if (hasArraySpace()) {
-            arrayList[size] = value;
-            size++;
-            return;
-        }
-        extendCapacity();
+        extendCapacityIfNecessary();
         arrayList[size] = value;
         size++;
     }
 
     @Override
     public void add(T value, int index) {
-        if (!hasArraySpace()) {
-            extendCapacity();
-        }
-        if (index < 0 || index > arrayList.length || index > size + 1) {
-            throw new ArrayListIndexOutOfBoundsException("Your index does not suit rules");
+        extendCapacityIfNecessary();
+        if (index >= 0 && index <= size) {
+            System.arraycopy(arrayList, index, arrayList, index + 1, size - index);
+            arrayList[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index.");
         }
-        System.arraycopy(arrayList, index, arrayList, index + 1, size - index);
-        arrayList[index] = value;
-        size++;
-
     }
 
     @Override
     public void addAll(List<T> list) {
         if (list != null) {
```

Comment:
it's better to use `for-each` loop here, in that way you won't need to check list for null